crates/core/src/dht/dht_store.rs

```diff
@@ -30,13 +30,17 @@ use crate::{dht::pending_validations::PendingValidation, state::StateWrapper};
 use chrono::{offset::FixedOffset, DateTime};
 use holochain_json_api::error::JsonResult;
 use holochain_persistence_api::error::PersistenceResult;
+use snowflake::ProcessUniqueId;
 use std::{
-    collections::{BTreeSet, VecDeque},
+    collections::{BTreeSet, HashMap, VecDeque},
     convert::TryFrom,
     sync::Arc,
     time::Duration,
 };
 
+/// A type for identifying holding attempts uniquely and by parent pending validation id
+pub type HoldAspectAttemptId = (ProcessUniqueId, ProcessUniqueId);
```

Comment:
it could be an improvement to give keys to the processuniqueid by making it a struct? for legibility

Reply:
I actually went down that path, but rustc requires you to do a bunch of stuff with clone/copy to structs that it just figures out when you use a tuple so I just kept it simple.